### ISSUE TITLE
swagger-codegen-maven-plugin overwrites files even when listed on .swagger-codegen-ignore

### ISSUE THREAD

**Issue by grootknol** [swagger-api/swagger-codegen#4444]
##### Description

swagger-codegen-maven-plugin always overwrites all generated files.

I'm generating server code for JAX-RS (`<language>jaxrs</language>`).

I need to change the *Impl files so I removed them from the `/target/generated-sources/swagger/` tree and placed them in my build path. The build works ok, but when if I regenerate the sources the codegen places a new copy in the generated-sources directory and my build fails because I have duplicate classes.

I added the generated files names to .swagger-codegen-ignore and made maven exclude that file from cleaning, but swagger-codegen-maven-plugin still generates the files.

##### Swagger-codegen version

Originally I was using swagger-codegen-maven-plugin version 2.2.2-SNAPSHOT, but the same issue happens on 2.2.1.

##### Command line used for generation

```
<plugin>
	<groupId>io.swagger</groupId>
	<artifactId>swagger-codegen-maven-plugin</artifactId>
	<version>2.2.1</version>
	<executions>
		<execution>
			<goals>
				<goal>generate</goal>
			</goals>
			<configuration>
				<inputSpec>${swagger.input}</inputSpec>
				<language>jaxrs</language> <!-- generate server stubs for JAX-RS -->
				<addCompileSourceRoot>true</addCompileSourceRoot>
				<modelPackage>swagger.model</modelPackage>
				<apiPackage>swagger.api</apiPackage>
				<configOptions>
					<sourceFolder>src/main/java</sourceFolder>
					<serializableModel>true</serializableModel>
				</configOptions>
				<addCompileSourceRoot>true</addCompileSourceRoot>
				<skipOverwrite>OrcamentoApiServiceImpl.java,PropostaApiServiceImpl.java</skipOverwrite>
				<environmentVariables>
					<!-- generate all models -->
					<models></models>
					<!-- generate all APIs -->
					<apis></apis>
					<!-- generate just the supporting files that are Java source code 
						(not project build files) -->
					<!-- <supportingFiles>NotFoundException.java,ApiException.java,ApiResponseMessage.java</supportingFiles> -->
					<supportingFiles></supportingFiles>
				</environmentVariables>
			</configuration>
		</execution>
	</executions>
</plugin>
<plugin>
	<artifactId>maven-clean-plugin</artifactId>
	<version>2.6.1</version>
	<configuration>
		<excludeDefaultDirectories>true</excludeDefaultDirectories>
		<filesets>
			<fileset>
				<directory>${project.build.directory}</directory>
				<excludes>
					<exclude>generated-sources/swagger/.swagger-codegen-ignore</exclude>
				</excludes>
			</fileset>
		</filesets>
	</configuration>
</plugin>
```
`mvn clean package`

**Comment by wing328** [comment]
@grootknol it could be related to https://github.com/swagger-api/swagger-codegen/issues/4452

**Comment by jimschubert** [comment]
@grootknol Can you share how you've defined your exclusion rules?

```
# Matches in same directory as .swagger-codegen-ignore as a shell glob
*Impl.java

# Matches all occurrences in all subdirectories
**/*Impl.java
```

**Comment by grootknol** [comment]
@jimschubert Sorry, I already fixed the problem. 

I had defined my exclusions as `*Impl.java`, as I understood it would exclude the files whenever they would be. Reading the comment in your post now, I guess I misunderstood and that rule would only exclude files in the same directory as `.swagger-codegen-ignore`.

My "fix" was to use the second exclusion rule in your post: `**/*Impl.java`.

Thank you for your response!
